src/main/java/org/opentripplanner/api/mapping/LegMapper.java

```diff
@@ -100,12 +110,73 @@ else if (domain.pathway) {
         );
         api.boardRule = domain.boardRule;
         api.alightRule = domain.alightRule;
+
+        api.pickupBookingInfo = toApi(domain.pickupBookingInfo);
+        api.dropOffBookingInfo = toApi(domain.dropOffBookingInfo);
+
         api.rentedBike = domain.rentedVehicle;
         api.walkingBike = domain.walkingBike;
 
         return api;
     }
 
+    private ApiBookingInfo toApi(BookingInfo info) {
+        if (info != null) {
+
+            return new ApiBookingInfo(
+                    toApi(info.getContactInfo()),
+                    toApi(info.bookingMethods()),
+                    toApi(info.getEarliestBookingTime()),
+                    toApi(info.getLatestBookingTime()),
+                    info.getMinimumBookingNotice(),
+                    info.getMaximumBookingNotice(),
+                    info.getMessage(),
+                    info.getPickupMessage(),
+                    info.getDropOffMessage()
+            );
+        }
+        else {
+            return null;
+        }
+    }
+
+    private ApiContactInfo toApi(ContactInfo info) {
+        if (info != null) {
+            return new ApiContactInfo(
+                    info.getContactPerson(),
+                    info.getPhoneNumber(),
+                    info.geteMail(),
+                    info.getFaxNumber(),
+                    info.getInfoUrl(),
+                    info.getBookingUrl(),
+                    info.getAdditionalDetails()
+            );
+        }
+        else {
+            return null;
+        }
+    }
+
+    private Set<String> toApi(EnumSet<BookingMethod> m) {
+        if (m != null) {
+            return m.stream()
+                    .map(Enum::toString)
```

Comment:
You should add explicit mapping for all enums. We might add/change the internal model and you might not want this change to propagate to the REST API.

Reply:
Done.